common/src/main/java/io/netty/util/internal/ReferenceCountUpdater.java

```diff
@@ -80,8 +65,7 @@ public final int refCnt(T instance) {
 
     public final boolean isLiveNonVolatile(T instance) {
         final int rawCnt = getRawRefCnt(instance);
-        // The "real" ref count is > 0 if the rawCnt is even.
-        return rawCnt == 2 || rawCnt == 4 || rawCnt == 6 || rawCnt == 8 || (rawCnt & 1) == 0;
+        return (rawCnt & 1) == 0;
```

Comment:
The original code set of comparison was a sort of cheap trick for the JIT compiler (and CPU) to have a super fast check on known common scenario (which clearly won't work as good for the `Chunk` use case - or with slicing or similar cases)
Try running https://github.com/netty/netty/pull/15452#issuecomment-3046874614 and see if it matters for your machine 
It has the most typical use cases for this super hot path